app/src/full/java/io/homeassistant/companion/android/sensors/HealthConnectSensorManager.kt

```diff
@@ -434,8 +440,12 @@ class HealthConnectSensorManager : SensorManager {
 
     override suspend fun checkPermission(context: Context, sensorId: String): Boolean {
         val healthConnectClient = getOrCreateHealthConnectClient(context) ?: return false
-        val result = healthConnectClient.permissionController.getGrantedPermissions().containsAll(requiredPermissions(sensorId).toSet())
-        return result
+        return try {
+            healthConnectClient.permissionController.getGrantedPermissions().containsAll(requiredPermissions(sensorId).toSet())
+        } catch (e: Exception) {
+            Log.e(TAG, "Unable to check permissions", e)
+            false
```

Comment:
This works, but will actively turn off the HC sensors once we reach the rate limit. Is that really desirable?

Update [checks `isEnabled`](https://github.com/home-assistant/android/blob/master/app/src/full/java/io/homeassistant/companion/android/sensors/HealthConnectSensorManager.kt#L195) -> `isEnabled` will check the permission and [pass the false value to the database](https://github.com/home-assistant/android/blob/master/common/src/main/java/io/homeassistant/companion/android/common/sensors/SensorManager.kt#L113) -> the database will [disable any enabled sensors found](https://github.com/home-assistant/android/blob/master/common/src/main/java/io/homeassistant/companion/android/database/sensor/SensorDao.kt#L165) while there is no permission.